src/compiler/checker.ts

```diff
@@ -17358,6 +17362,17 @@ namespace ts {
                     }
                 }
             }
+            if (depth >= 5 && getObjectFlags(type) && ObjectFlags.Reference && !!(type as TypeReference).node) {

```

Comment:
So this is to handle cases where a type reference doesn't have a symbol? Such as tuple types? If so, minor point, feels like this should all be under a common test for `depth >= 5 && type.flags & TypeFlags.Object`.

Reply:
Aye. A `Reference` is, definitionally, required to be an `Object`.